### PR TITLE
(feat) - remove presence validation of password

### PR DESCRIPTION
**Pull request by amit-choudhary**

### CODE REVIEW

**Comment by amit-choudhary** [comment]
@akshaysmurthy Please review.

File: lib/promostandards/client.rb [hunk 1]
```diff
@@ -14,11 +14,11 @@ class Client
 
     PRIMARY_IMAGE_PRECEDENCE = ['1006', ['1007', '1001', '2001'], ['1007', '1001'], '1007', ['1001', '2001'], '1001', '1003']
 
-    def initialize(access_id:, password:, product_data_service_url:, media_content_service_url:)
-      @access_id = access_id
-      @password = password
-      @product_data_service_url = product_data_service_url
-      @media_content_service_url = media_content_service_url
+    def initialize(config = {})
```
Comment (akshaysmurthy) [inline lib/promostandards/client.rb]: We could still use keyword arguments - 
```suggestion
    def initialize(access_id:, password: nil, product_data_service_url:, media_content_service_url: nil)
```

While you are at it, can you also set the default value for `media_content_service_url` as suggested above? I forgot to do this earlier.

Comment (amit-choudhary) [reply lib/promostandards/client.rb]: Didn't get you.

Comment (akshaysmurthy) [reply lib/promostandards/client.rb]: Which part? 
1. We can still use keyword arguments (and not a hash argument `config`) by setting default values for `password` and `media_content_service_url`. I've added a code suggestion above.
2. I had missed making `media_content_serive_url` optional. But this is also there in the code suggestion.
These would essentially make `password` and `media_content_service_url` optional and keeping the others mandatory.

**Comment by akshaysmurthy** [comment]
@amit-choudhary There was a problem in the last commit. I've fixed the same in https://github.com/promohunt/promostandards-client/commit/3a4250688912815f3b53f8fce8a58695a5c6e145 
Please run tests on the dev machine before pushing :)
I'll set up CI for this project today.

**Comment by akshaysmurthy** [comment]
[PromoStandards: Password should be an optional field](https://trello.com/c/jU2vdXEV/1027-promostandards-password-should-be-an-optional-field)

**Pull request merged**